src/main/java/org/elasticsearch/script/ScriptService.java

```diff
@@ -579,6 +609,16 @@ public static void writeTo(ScriptType scriptType, StreamOutput out) throws IOExc
                 out.writeVInt(INLINE_VAL); //Default to inline
             }
         }
+
+        @Override
+        public String toString() {
+            //ScriptModes looks for 'dynamic' rather than 'inline' when reading settings, that's the terminology
+            //we always used with users, although we prefer inline internally
+            if (this == INLINE) {
```

Comment:
we are talking about  toString method?! No way we add a distinction in there... if we pass this to the user then fix it in the rest layer